### ISSUE TITLE
Segfault when switching tool after dragging text while typing

### ISSUE THREAD

**Issue by mmahmoudian** [flameshot-org/flameshot#1421]
**Flameshot version**
Flameshot v0.9.0
Compiled with Qt 5.15.2

**Describe the bug**
When typing text while dragging the textbox (meaning the left-click is constantly pressed), when then you release the left-click and click on any tool, Flameshot crashes:

![Peek 2021-03-03 17-09](https://user-images.githubusercontent.com/390889/109826946-fcbf9880-7c43-11eb-8722-b5ff51686274.gif)

Also this is what I get in terminal when starting `flameshot` and then starting `flameshot gui` in another terminal.

```
QSettings::value: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QPainter::begin: Paint device returned engine == 0, type: 2
QPainter::setRenderHint: Painter must be active to set rendering hints
QPainter::setCompositionMode: Painter not active
QPainter::translate: Painter not active
QPainter::setPen: Painter not active
QPainter::setBrush: Painter not active
QPainter::setBrush: Painter not active
[1]    159200 segmentation fault (core dumped)  flameshot
```

**To Reproduce**
1. `flameshot gui`
2. select the text tool, left-click on the screen and don't release the left-click (you can drag the mouse or keep it still)
3. start typing
4. release the left-click
5. click on any other tool

**Expected behavior**

I expect that Flameshot does not segfault

**System Information**
```
System:
  Host: Chrstphr Kernel: 5.10.18-1-MANJARO x86_64 bits: 64
  Desktop: KDE Plasma 5.21.1 Distro: Manjaro Linux
Graphics:
  Device-1: Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics
  driver: i915 v: kernel
  Device-2: NVIDIA GP106 [GeForce GTX 1060 6GB] driver: nvidia v: 460.39
  Display: x11 server: X.Org 1.20.10 driver: loaded: modesetting,nvidia
  unloaded: nouveau resolution: 1: 1080x1920~60Hz 2: 1080x1920~60Hz
  3: 1920x1080~60Hz
  OpenGL: renderer: GeForce GTX 1060 6GB/PCIe/SSE2 v: 4.6.0 NVIDIA 460.39
```

**Comment by borgmanJeremy** [comment]
Early analysis shows that the issue is m_activeTool is nullptr here: https://github.com/flameshot-org/flameshot/blob/f54538ee1ea1ae95faf3d00c14813966439094f7/src/widgets/capture/capturewidget.cpp#L834

**Comment by borgmanJeremy** [comment]
Will be solved with #1422 .
